src/uct/ib/dc/dc_mlx5.c

```diff
@@ -566,14 +566,19 @@ static ucs_status_t
 uct_dc_mlx5_init_rx(uct_rc_iface_t *rc_iface,
                     const uct_rc_iface_common_config_t *rc_config)
 {
-    uct_ib_mlx5_md_t *md               = ucs_derived_of(rc_iface->super.super.md, uct_ib_mlx5_md_t);
-    uct_dc_mlx5_iface_config_t *config = ucs_derived_of(rc_config, uct_dc_mlx5_iface_config_t);
-    uct_dc_mlx5_iface_t *iface         = ucs_derived_of(rc_iface, uct_dc_mlx5_iface_t);
+    uct_ib_mlx5_md_t *md                 = ucs_derived_of(rc_iface->super.super.md,
+                                                          uct_ib_mlx5_md_t);
+    uct_dc_mlx5_iface_config_t *config   = ucs_derived_of(rc_config,
+                                                          uct_dc_mlx5_iface_config_t);
+    uct_dc_mlx5_iface_t *iface           = ucs_derived_of(rc_iface,
+                                                          uct_dc_mlx5_iface_t);
     struct ibv_srq_init_attr_ex srq_attr = {};
+    int devx_srq_supported               = uct_rc_mlx5_is_devx_srq_supported(&iface->super,
+                                                                             md, 1);
```

Comment:
i think it would be simpler to remove uct_rc_mlx5_is_devx_srq_supported and instead test of the specific md_flags in every place it's used. for example, we can eliminate UCT_RC_MLX5_TM_ENABLED check and QP type check.